packages/webdriverio/src/commands/element/click.ts

```diff
@@ -83,77 +83,62 @@ import type { Button } from '../../utils/actions/index.js'
  */
 export async function click(
     this: WebdriverIO.Element,
-    options?: ClickOptions
+    options: Partial<ClickOptions> = {}
 ) {
-    if (typeof options === 'undefined') {
-        return this.elementClick(this.elementId)
```

Comment:
What's the incentive to move away from the basic click command?

Reply:
> This meant that the click command did not use the workaround that was implemented ago by default, which in fact it should.

The WebDriver protocol does the "scroll into view" work for us:

> [Scroll into view](https://www.w3.org/TR/webdriver2/#dfn-scrolls-into-view) the element’s [container](https://www.w3.org/TR/webdriver2/#dfn-container).

This has been working reliably.

Reply:
Sure, just run this snippet:
```js
    await browser.url('https://webdriver.io');
    await $('div*=Watch Talks about WebdriverIO').scrollIntoView();
    await $('a[href="https://github.com/webdriverio"]').click();
    await browser.pause(2000);
```
The workaround we have in the click command for this does not work because it works differently by default compared to when passing in an object. I think we should use one method, not two.

Reply:
In order to get our tests with similar behaviour working in our project we have the following workaround which I want to eliminate:
```js
const elemToClick = $('example')
await elemToClick.scrollIntoView({ block: 'center' }) // first scroll the element into view (NOT the top since often times there's a header that is in the way)
await elemToClick.click()
```
Other possible workarounds would be:
```js
const elemToClick = $('example')
await elemToClick.click({}) // since it now sees an element, it will skip the element click and will correctly apply the workaround
```
```js
browser.overwriteCommand('click', async function (originClick, options) {
  originClick(options || {})
}, true);
const elemToClick = $('example')
await elemToClick.click() // can use the click like you normally would but instead of undefined, it passes an empty object so that the workaround is applied
```

Reply:
I have also considered about applying the same workaround to the elementClick but I want to get rid of the difference in behaviour and provide a robust click that behaves the same. Whether or not I pass in an object or not should not matter.
```js
if (typeof options === 'undefined') {
  try {
    return this.elementClick(this.elementId)
  } catch {
    await this.scrollIntoView({ block: 'center', inline: 'center' })
    return this.elementClick(this.elementId)
  }
}      
```